kebechet/managers/update/update.py

```diff
@@ -154,18 +158,36 @@ def _get_all_packages_versions() -> dict:
             raise DependencyManagementError(f"Failed to load Pipfile.lock file: {str(exc)}") from exc
 
         result = {}
-        for package_name, package_info in pipfile_lock_content['default'].items():
-            result[package_name.lower()] = {
-                'dev': False,
-                'version': package_info['version'][len('=='):]
-            }
-
-        for package_name, package_info in pipfile_lock_content['develop'].items():
-            result[package_name.lower()] = {
-                'dev': False,
-                'version': package_info['version'][len('=='):]
-            }
+        try:
+            for package_name, package_info in pipfile_lock_content['default'].items():
+                result[package_name.lower()] = {
+                    'dev': False,
+                    'version': package_info['version'][len('=='):]
+                }
 
+            for package_name, package_info in pipfile_lock_content['develop'].items():
+                result[package_name.lower()] = {
+                    'dev': False,
+                    'version': package_info['version'][len('=='):]
+                }
+            # Close git as a source issues.
+            self.sm.close_issue_if_exists(_ISSUE_UNSUPPORTED_PACKAGE, f"Issue closed as no packages use git as a \
+                        source anymore. Related SHA - {self.sha}")
+        except KeyError as exc:
```

Comment:
To reduce complexity, you don't need to use try-except, but directly do the check `if "git" in package_info`. Also, the exception here is not re-raised, the code will continue as no error would occur. Is that intentional?

Reply:
```
            for package_name, package_info in pipfile_lock_content['default'].items():
                if 'git' in package_info:
                      call_method_that_creates_issue()
            result[package_name.lower()] = {
                    'dev': False,
                    'version': package_info['version'][len('=='):]
                }
```
Is this how you think would be better? And just let the error happen?

Reply:
I have removed the try block :)